meetings/2024-10-02.md

```diff
@@ -0,0 +1,109 @@
+# W3C Solid Community Group: Weekly
+
+* Date: 2024-10-02T14:00:00Z
+* Call: https://meet.jit.si/solid-cg
+* Chat: https://matrix.to/#/#solid_specification:gitter.im
+* Repository: https://github.com/solid/specification
+* Status: Published
+
+
+## Present
+* [Michiel de Jong](https://michieldejong.com)
+* Jesse Wright
+* [elf Pavlik](https://elf-pavlik.hackers4peace.net)
+* [Virginia Balseiro](https://virginiabalseiro.com/#me)
+* Grace Elcock (observer)
+* [Sarven Capadisli](https://csarven.ca/#i)
+* [Rahul Gupta](https://cxres.pages.dev/profile#i)
+* Damon
+* Pierre-Antoine Champin
+
+---
+
+## Announcements
+
+* SC: W3C encourage people to join working groups, for instance as invited experts. It's important for continuity.
+
+### Meeting Guidelines
+* [W3C Solid Community Group Calendar](https://www.w3.org/groups/cg/solid/calendar).
+* [W3C Solid Community Group Meeting Guidelines](https://github.com/w3c-cg/solid/blob/main/meetings/README.md).
+* No audio or video recording, or automated transcripts without consent. Meetings are transcribed and made public. If consent is withheld by anyone, recording/retention must not occur.
+* Join queue to talk.
+* Topics can be proposed at the bottom of the agenda to be discussed as time allows. Make it known if a topic is urgent or cannot be postponed.
+
+### Participation and Code of Conduct
+* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/)
+* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Conduct](https://www.w3.org/policies/code-of-conduct/)
+* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
+* If this is your first time, welcome! please introduce yourself.
+
+
+### Scribes
+* Michiel de Jong
+* Sarven Capadisli
+
+### Introductions
+*
+
+---
+
+## Topics
+
+### Solid Interop in Practice
+
+URL: https://forum.solidproject.org/t/solid-interop-in-practice/7701
+
+* eP: we mentioned this topic two weeks ago, this is a write-up by Rosano, who interviewed multiple Solid app developers. My main take-away is people find especially client-client interop hard to work with
+* eP: General comments on JS, there is a lot of work there but it's not specific to Solid.
```

Comment:
I am not a developer but have been following Solid on Github for years. This conversation was interesting to me, so I engaged with ChatGPT on it to learn more about shapes and what the problem might be.  I thought the answers were interesting from the perspective of an outsider. I fed the entire conversation on this thread to chat. 

~~~~~~~~~~
You said:
Do you know what they mean by shapes?
~~~~~~~~~~~~
ChatGPT said:
ChatGPT
In the context of Solid (Social Linked Data) and RDF (Resource Description Framework), "shapes" refer to a way of defining the structure and constraints of RDF data. Specifically, shapes are part of a technology known as SHACL (Shapes Constraint Language), which is used to validate RDF data.

A shape defines expectations for the structure of data, such as what types of properties an RDF resource should have, the types of those properties, and other constraints like cardinality (e.g., how many times a property can occur). For instance, a "Person" shape might specify that a person has a name, an age, and an email, and that the name must be a string, the age an integer, and the email a valid email address.

Shapes are useful for ensuring interoperability between different applications that rely on RDF data. They provide a way to check if data conforms to a particular structure, making it easier for different systems or apps to work together by agreeing on a common "shape" of the data. In Solid, this could be used to make sure that data in a pod (personal online datastore) conforms to the expected structure that an application needs to function properly.

For example, if two apps share the same shape for an event or contact data, they can more easily read, write, and interpret each other's data, facilitating "interoperability" between Solid apps.

Does that clarify it?
~~~~~~~~~~
You said:
What seems to be the problem in this conversation re: shapes?
~~~~~~
ChatGPT said:
ChatGPT
The conversation around "shapes" in the Solid meetings seems to focus on a few key challenges:

1. Lack of Standardization and Collaboration on Shapes
There is a recurring theme in the discussion where participants note that app developers often have to create their own shapes, which adds a burden. The expectation seems to be that developers should not be left to define shapes independently. Instead, developers working in the same domain should collaborate to create a shared set of shapes (a version 1, for instance). This would reduce fragmentation and ensure greater interoperability. However, this kind of collaboration is not happening effectively, and there seems to be no easy or structured way for developers to agree on common shapes.

2. Domain-Specific vs. Generic Shapes
The conversation also points to an issue with defining what should be domain-specific shapes versus more generic ones. For example, there are certain kinds of data like calendar events or architectural blueprints that may require specific shapes unique to that domain. Participants argue about whether it’s best to create highly specialized shapes or to focus on more generic shapes that could work across different contexts. This balance between domain-specific needs and more general-purpose shapes seems unresolved.

3. Shape Discovery and Reuse
Another issue is the challenge of finding and reusing existing shapes. Some participants mention existing shape repositories, like the one on GitHub, but it's unclear how well these are being used or integrated. The lack of a clear process for discovering, reusing, or even sharing shapes seems to hinder interoperability. This suggests that there’s a lot of duplication of effort and inefficiency because developers might not know what shapes already exist or how to integrate them into their projects.

4. Composability and Nesting of Shapes
The conversation also touches on the technical complexity of making shapes more composable and nested. Participants mention the need to support more advanced use cases, like nesting shapes (for example, a shape within a shape). This could lead to complex requirements around access control and how different datasets interact with each other, which isn't being adequately addressed by current tools or protocols.

5. Process of Iterating on Shapes
Lastly, there seems to be an underlying issue with the process itself—how shapes should be iterated on, agreed upon, and updated. Some participants believe that there needs to be a better framework for "process and iterating," which would help standardize the work around shape definitions and make the development of shapes more systematic. The absence of this process means that current efforts are fragmented, leading to a slower path toward interoperability.

In summary, the conversation highlights the difficulty in establishing common shapes, the challenge of balancing generality with specificity, the lack of effective discovery and reuse mechanisms, and the need for a clearer process for shaping collaboration in Solid. The current approach is ad-hoc, which leads to fragmented solutions that undermine the goal of interoperability between Solid applications.
~~~~~~~
You said:
What would you recommend the approach be?
~~~~~~~~

ChatGPT said:
ChatGPT
To address the challenges discussed in the Solid conversation around "shapes," a structured approach could help foster better collaboration, standardization, and interoperability. Here are some key recommendations:

1. Establish a Centralized Shape Registry
A centralized and well-maintained Shape Repository (or registry) should be created, where all shared shapes are published, documented, and easily accessible to developers. This would include:

Versioning: Allowing different versions of shapes to evolve over time while maintaining backward compatibility.
Metadata: Clearly labeling the domain, purpose, and constraints of each shape (e.g., for calendar events, architectural blueprints, etc.).
Discoverability: Ensuring developers can easily search for existing shapes based on domain, use case, or data type, avoiding duplication of effort.
The key is to make it easy for developers to reuse existing shapes and contribute back, reducing fragmentation.

2. Define a Process for Shape Standardization
Introduce a shape standardization process similar to how W3C or other technical bodies define standards:

Community Collaboration: Establish working groups (or task forces) around specific domains (e.g., social networking, event planning, healthcare) where developers working on similar applications can come together to define shapes.
Consensus-Based Approach: Allow developers to propose shapes, but ensure a community review process before finalizing them. This prevents isolated or redundant shape definitions.
Interoperability Testing: Before shapes are added to the central registry, run tests to ensure they work in different applications or contexts to validate interoperability.
This would formalize how shapes are defined, refined, and adopted, reducing the burden on individual developers.

3. Encourage Reusability Through Modular Shapes
To balance between generic and domain-specific needs, promote modular, composable shapes. Shapes should be designed in a way that:

Core/Generic Shapes: Define core shapes that represent general concepts (e.g., person, event) with a minimal set of constraints.
Extensions/Domain-Specific Shapes: Allow domain-specific shapes (e.g., medical appointment) to build on top of these core shapes by adding specific constraints or properties. This will enable reuse across domains while still supporting domain-specific requirements.
By creating building blocks of shapes, developers can customize existing ones for their specific needs without starting from scratch.

4. Tooling for Shape Validation and Generation
Provide tools that make using and creating shapes easier:

Shape Validation Tools: Offer easy-to-use tools that allow developers to validate RDF data against a shape. This will give developers confidence that their data conforms to the expected structure and can be shared across applications.
Shape Generators: Develop tools that generate shapes from existing RDF data or ontology. For example, if a developer already has a data model in use, they could generate the corresponding shape and refine it collaboratively.
These tools would lower the technical barrier to adopting shapes and encourage broader participation from developers who are not RDF or SHACL experts.

5. Facilitate Domain-Specific Communities
Encourage the formation of domain-specific working groups within the Solid community. These groups can:

Agree on Shared Shapes: Ensure collaboration among developers working on similar applications to define and agree on common shapes for their domain (e.g., education, finance, healthcare).
Shape Governance: Create a lightweight governance model for how shapes are proposed, reviewed, and iterated on, similar to how standards bodies like W3C operate.
Shape Review Process: Establish a mechanism for feedback and improvement of existing shapes over time based on real-world use cases.
Facilitating these communities would allow domain experts to bring their knowledge into the process and align on best practices.

6. Develop Use Case-Driven Shape Proposals
Instead of proposing shapes in a vacuum, ensure they are use-case driven:

Real-World Applications: Shape proposals should come directly from the needs of real-world applications or communities that are using Solid. This ensures that shapes are relevant and directly tied to practical challenges developers face.
Pilot Projects: Run pilot projects with willing participants where new shapes are proposed and tested in real applications before they are made part of the standard set.
This approach would ensure that the shapes being developed solve real interoperability issues in practice, rather than being theoretical.

7. Shape Composition and Relations
Address the need for composability and nested shapes by encouraging:

Shape Inheritance: Allow shapes to inherit properties from other shapes. For example, a "MedicalAppointment" shape could inherit from a generic "Event" shape but add specific fields for doctor’s name, prescription, etc.
Context-Aware Shapes: Develop patterns for how shapes are applied depending on the context, especially for access control. This is critical for sensitive data (e.g., health records) where different data consumers might need different levels of access.
Building these patterns into Solid’s shape system would make them more flexible and powerful for real-world use.

8. Shape-First Approach for Interoperability
Finally, promote a shape-first approach to interoperability:

Shared Shape Libraries: Encourage developers to start by defining or choosing shapes before developing applications. If apps are built around shared shapes, interoperability becomes easier to achieve.
Shape Validation as Part of App Development: Make shape validation an integral part of app development, so developers ensure compliance with agreed-upon structures early in the development process.
This approach would shift the focus toward more proactive alignment on data structures, leading to greater interoperability across Solid apps.

By following these recommendations, the Solid community can move toward a more standardized, collaborative approach to defining and using shapes, addressing the current fragmentation and interoperability challenges.

Reply:
@paulduplantis — 

I suggest you put this comment on a higher level, either as [a new issue](https://github.com/solid/specification/issues/) since it doesn't seem like your comments are tied to the meeting for which these are the minutes, or as a comment [on this PR](https://github.com/solid/specification/pull/687) instead of as a comment on a change suggestion on this PR. 

When you do this, I strongly suggest that you do not put such large chunks of the content into code blocks, as they are very hard to read, requiring manual side-scrolling for each VERY long line of text. I suggest you use the `>` character to indicate quoted sections, using two of them `>>` for your prompts and one of them `>` for ChatGPT's responses.